numpy/ma/core.py

```diff
@@ -2968,8 +2968,8 @@ def __array_wrap__(self, obj, context=None):
                 np.copyto(result, fill_value, where=d)
                 # Update the mask
                 if m is nomask:
-                    if d is not nomask:
-                        m = d
+                    if any(d):
+                        m = True if all(d) else d
```

Comment:
I doubt it is OK to set the mask of a full array of True to True. At the very least it would break assignments, since they check nomask/False explicitly but not allmasked or whatever you would call it.